docs/architecture/adr/ADR043-k8s-resource-access.md

```diff
@@ -0,0 +1,783 @@
+# ADR043: Kubernetes resource access rules
+
+## Status
+
+Pending
+
+## Context
+
+Several different levels of access are required within a kubernetes cluster. The
+GSP uses role based access control so these levels are granted to users and
+groups via roles.
+
+## Decision
+
+We will create two levels of access within each namespace:
+
+* Operator
+* Auditor
+
+The Operator is a relatively permissive read-write role within the namespace.
+Developers working on branches that are not part of the release process may be
+granted this role in certain namespaces. This is also the role the in-cluster
+concourse team for each namespace will be granted.
+
+The Auditor will be given to all authenticated users in the cluster. This should
+allow for debugging of issues and incidents and basic remedial actions without
+needing formal escalation procedures.
+
+The complete list of resource permissions is given in Appendix A.
+
+## Consequences
+
+* Site Reliability Engineers (SRE) will be able to effectively operate the GSP
+  while reducing the need to escalate to "admin" level access.
+* service team members (e.g. devs) will be able to iterate changes quickly via
+  branch-based development and/or `kubectl apply` level access to specific
+  namespaces.
+* service team members will not be able to interfere with other devs'
+  namespaces, "system" namespaces or "prod" namespaces.
+* SREs and service team members will not be able to cause major production
+  service outages. Degradation will be possible through deletion of certain
+  resources but that will be easily fixable by running the appropriate
+  deployment pipeline.
+
+
+## Appendix A: Permissions rules
+
+---
+apiVersion: rbac.authorization.k8s.io/v1beta1
+kind: ClusterRole
+metadata:
+  name: auditor
+rules:
+- apiGroups: [""]
+  resources:
+  - configmaps
+  - endpoints
+  - limitranges
+  - pods
+  - podtemplates
+  - resourcequotas
+  - serviceaccounts
+  - services
+  verbs:
+  - delete
+  - get
+  - list
+  - watch
+- apiGroups: [""]
+  resources:
+  - events
+  - namespaces
+  - nodes
+  - persistentvolumeclaims
+  verbs:
+  - get
+  - list
+  - watch
+- apiGroups: [""]
+  resources:
+  - secrets
+  verbs:
+  - delete
+  - list
+  - watch
```

Comment:
I don't think we should allow `watch` on Secrets - doesn't it let you get the contents?  I'm not sure of any circumstance where `watch` *should* be allowed and I am concerned that it might allow getting things that it shouldn't allow.